### PR TITLE
Feat: Add support with AWS access key cred

### PR DESCRIPTION
**Pull request by GiliFaroEnv0**
### Issue & Steps to Reproduce / Feature Request
Solve #232 
### Solution

Add support to AWS access key credentials

### CODE REVIEW

File: client/aws_credentials.go
Comment (HeverFarber) [inline client/aws_credentials.go:43]: It could break the usage of customers.

I agree this default is ugly - but if someone uses this resource. it will force him to commit change before new deploy. 

maybe use the default with warning message

File: client/aws_credentials_test.go
Comment (liranfarage89) [inline client/aws_credentials_test.go:53]: can't you use the const types you created in `model.go`?

File: client/aws_credentials.go
Comment (liranfarage89) [reply client/aws_credentials.go:43]: @HeverFarber but I think it is still backward's compatible... the old usage still requires `arn`& `externa_id`, and the `Type` was assigned under the hood.

**Review by liranfarage89 — APPROVED** [review]
LGTM - but let's wait for @HeverFarber response before merging it.

File: client/aws_credentials.go
Comment (GiliFaroEnv0) [reply client/aws_credentials.go:43]: @HeverFarber  as @liranfarage89 said Its backward's compatible.

Comment (HeverFarber) [reply client/aws_credentials.go:43]: We should keep backward's compatible. for the who doesn't add type before. we can add it instead of him

Comment (liranfarage89) [reply client/aws_credentials.go:43]: but the type was not exposed before that...

Comment (HeverFarber) [reply client/aws_credentials.go:43]: Correct - so we added it for the user. so the user didn't add it to his resource.

Now when we removed it - the resource without this type - and when we will send it to the server we will get an error because type is missing

Comment (liranfarage89) [reply client/aws_credentials.go:43]: the Type was not a part of the resource either way, it's computed

Comment (eranelbaz) [reply client/aws_credentials.go:43]: I don't think the user will need to approve a change since the `Type` isn't part of the schema,
and the removal is in the `client` but @GiliFaroEnv0 is calculating the type on runtime in the "resource" side and sends it to the `client` here 

https://github.com/env0/terraform-provider-env0/blob/0ea2c1771e18d41948de435cd344a0ebe548b868/env0/resource_aws_credentials.go#L72-L92

and you guys can check it, run `terraform apply` using `main`, then checkout this branch and run `apply` again and check if there is any change

Comment (GiliFaroEnv0) [reply client/aws_credentials.go:43]: OK it works I merge IT 👍🏿 
bors r+

**Pull request merged**
